ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -683,6 +708,7 @@ private void processGetBlobResponse(GetRequestInfo getRequestInfo, GetResponse g
               // this is a successful response and one that completes the operation regardless of whether the
               // success target has been reached or not.
               chunkCompleted = true;
+              retainExceptionOnSuccess = true;
```

Comment:
Alright - although such a situation would be a programming error. Not sure if you want something that masks that.